zerver/tests/test_markdown.py

```diff
@@ -1032,16 +1032,23 @@ def test_realm_patterns(self) -> None:
 
         self.assertEqual(converted, '<p><a href="https://trac.example.com/ticket/ZUL-123">#ZUL-123</a> was fixed and code was deployed to production, also <a href="https://trac.example.com/ticket/zul-321">#zul-321</a> was deployed to staging</p>')
 
-        def assert_conversion(content: str, should_have_converted: bool=True) -> None:
+        def assert_conversion(content: str, should_have_converted: bool=True, check_topic: bool=True) -> None:
             converted = markdown_convert(content, message_realm=realm, message=msg)
             converted_topic = topic_links(realm.id, content)
+            # Some tests modify the domain as well, so we have two different asserts for each case.
             if should_have_converted:
-                self.assertTrue('https://trac.example.com' in converted)
+                self.assertTrue('https://' in converted)
+                self.assertTrue('trac.example.com' in converted)
```

Comment:
This obviously isn’t a real vulnerability in the context of tests, but GitHub code scanning [doesn’t like this pattern](https://github.com/github/codeql/blob/main/python/ql/src/Security/CWE-020/IncompleteUrlSubstringSanitization.ql), which seems like a good reason to avoid it if it’s not too hard.